src/main/java/com/bobocode/blyznytsia/bibernate/persister/EntityPersisterImpl.java

```diff
@@ -0,0 +1,162 @@
+package com.bobocode.blyznytsia.bibernate.persister;
+
+import static com.bobocode.blyznytsia.bibernate.util.EntityUtil.getEntityIdValue;
+import static com.bobocode.blyznytsia.bibernate.util.EntityUtil.getEntityNonIdValues;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildDeleteStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildInsertStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildSelectStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildUpdateStatement;
+
+import com.bobocode.blyznytsia.bibernate.exception.PersistenceException;
+import com.bobocode.blyznytsia.bibernate.lambda.StatementConsumer;
+import com.bobocode.blyznytsia.bibernate.lambda.StatementFunction;
+import com.bobocode.blyznytsia.bibernate.util.EntityUtil;
+import com.bobocode.blyznytsia.bibernate.util.ResultSetMapper;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+public class EntityPersisterImpl implements EntityPersister {
+
+  Connection connection;
+  ResultSetMapper resultSetMapper;
+
+  public EntityPersisterImpl(Connection connection, ResultSetMapper resultSetMapper) {
+    this.connection = connection;
+    this.resultSetMapper = resultSetMapper;
+  }
+
+  @Override
+  public <T> List<T> findAll(Class<T> entityType, Field key, Object value) {
```

Comment:
As for me `findAll` means `get all records from DB`, so the name of the method is a bit confusing, probably  `findBy` would be better ...
By the way is it possible to get all records without any search criteria in current implementation?

Reply:
Agreed, the naming's a bit confusing, renamed it to `findAllBy`.
Regarding the second question - currently, it's not supported. The only use case for this method right now probably would be searching by foreign keys.